util/mkdef.pl

```diff
@@ -874,6 +874,10 @@ sub do_defs
 			next if(/\#define/);
 
 			# Reduce argument lists to empty ()
+			# get rid of any existing empty argument lists
+			while(/\(\)/) {
+				s/\(\)//gs;
+			}
 			# fold round brackets recursively: (t(*v)(t),t) -> (t{}{},t) -> {}
 			while(/\(.*\)/s) {
 				s/\([^\(\)]+\)/\{\}/gs;
```

Comment:
Re perl code, I did not understand what you meant. I hacked the perl. I do not know perl, so I hacked it based on similarity to other languages which I do know, C, bash script, Ruby, etc. I stopped hacking the perl when opensll build succeeded.

If a perl expert can find a better fix, please do by all means...